src/main/java/de/tum/cit/aet/artemis/buildagent/service/BuildJobManagementService.java

```diff
@@ -176,6 +177,13 @@ public CompletableFuture<BuildResult> executeBuildJob(BuildJobQueueItem buildJob
                 }
                 else {
                     finishBuildJobExceptionally(buildJobItem.id(), containerName, e);
+                    if (e instanceof TimeoutException) {
+                        String msg = "Timed out after " + buildJobTimeoutSeconds + " seconds. "
+                                + "This may be due to an infinite loop or inefficient code. Please review your code for potential issues. "
+                                + "If the problem persists, contact your instructor for assistance. (Build job ID: " + buildJobItem.id() + ")";
```

Comment:
Build logs cannot be read by students. Only instructors and admins have access. I suggest to change the message here and instead write that instructors should check for short and strict timeouts in the tests to prevent infinite loops in the student code